### DISCUSSION TITLE
Cartesian velocity constraints

### DISCUSSION THREAD

**Discussion by thettasch** [tesseract-robotics/tesseract_planning#402] [Q&A] [answered]
Is there a way to constrain the Cartesian velocity of the TCP in some way? I'm trying to plan a motion where the tool needs to travel at a constant speed across the workpiece, however so far I have been unable to find an obvious way to do this. I'd appreciate any advice/pointers to similar examples.

**Comment by gavanderhoorn** [comment, marked as answer]
I believe there is an implementation of this available in the `ros-industrial-consortium/scan_n_plan_workshop` (contributed in/by https://github.com/ros-industrial-consortium/scan_n_plan_workshop/pull/28).

**Reply by thettasch** [reply]
Thank you! This is exactly what I was looking for.

**Comment by rjoomen** [comment]
@Levi-Armstrong what do you think about adding this to core tesseract?

**Reply by marip8** [reply]
This has been on my to-do list for a little while. It seems to work reasonably well, but probably needs a review to make sure everything is computed correctly. The only bug that I know about currently is that it doesn't work with trajectories of kinematic trees, specifically robot with external positioner
